### PR TITLE
feat(OneTouchTV): add presence

### LINKED ISSUE #7646: onetouchtv | onetouchtv.me
### Website name

onetouchtv.me

### Website URL

https://onetouchtv.me/

### Website logo

https://prnt.sc/qZdD2qfu35eN

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

OneTouch TV is the hit Drama & Movie & TV app . OneTouch TV has it all in a single app. You can stream all your favorite movies & drama & TV for free.

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #7646
## Description 

Added the presence
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/O/OneTouchTV/metadata.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"thumbnail": "https://i.imgur.com/VGiNeKD.png",
+	"color": "#000000",
+	"category": "videos",
+	"tags": [
+		"tv",
+		"movie",
+		"serie"
+	],
+	"iframe": true,
+	"iFrameRegExp": ".*"
```
Comment (DarkVillager) [inline websites/O/OneTouchTV/metadata.json:23]: I'll look into this, cuz i think its possible to change it.

File: websites/O/OneTouchTV/presence.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		case pathname.includes("/detail/"): {
+			const title =
+					document
+						.querySelector("title")
+						.textContent.replace(/ - OneTouch TV/gm, "")
+						.trim() === ""
+						? document
+								.querySelector("body > flt-glass-pane")
+								.shadowRoot.querySelector(
+									"flt-scene-host > flt-scene > flt-transform > flt-offset > flt-offset > flt-clip > flt-clip-interior > flt-offset > flt-clip > flt-clip-interior > flt-clip > flt-clip-interior > flt-offset > flt-offset > flt-clip > flt-clip-interior > flt-offset:nth-child(2) > flt-picture:nth-child(3) > flt-canvas > flt-paragraph:nth-child(3) > flt-span:nth-child(1)"
```
Comment (DarkVillager) [inline websites/O/OneTouchTV/presence.ts:71]: This can't be shortened without making it unstable

File: websites/O/OneTouchTV/presence.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+								)?.textContent
+						: document
+								.querySelector("title")
+								?.textContent.replace(/ - OneTouch TV/gm, ""),
+				episode = title?.includes("Episode");
```
Comment (DarkVillager) [inline websites/O/OneTouchTV/presence.ts:76]: I will change this.

File: websites/O/OneTouchTV/presence.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
+const presence = new Presence({
+	clientId: "1160179473044476017",
+});
+
+class iframeData {
```
Comment (Bas950) [inline websites/O/OneTouchTV/presence.ts:5]: probs want to use interface here

File: websites/O/OneTouchTV/presence.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	timestamp: Math.floor(Date.now() / 1000),
+};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/v04o7nQ.png",
+			startTimestamp: timestampCheck.timestamp,
+		},
+		{ href, pathname } = document.location;
+	let hash: string = href;
```
Comment (DarkVillager) [inline websites/O/OneTouchTV/presence.ts:30]:
```suggestion
		{ href, pathname } = document.location,
	hash = href;
```

File: websites/O/OneTouchTV/iframe.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+	switch (true) {
+		case !!video: {
+			iframe.send({
+				paused: video.paused,
+				duration: video.duration,
+				currentTime: video.currentTime,
+			});
+			break;
+		}
+	}
```
Comment (theusaf) [inline websites/O/OneTouchTV/iframe.ts:15]: Strange use of switch, could be an if statement.

File: websites/O/OneTouchTV/metadata.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "638080361179512853",
+		"name": "dark_ville"
+	},
+	"service": "OneTouchTV",
+	"description": {
+		"en": "OneTouch TV is the hit Drama & Movie & TV app . OneTouch TV has it all in a single app. You can stream all your favorite movies & drama & TV for free."
+	},
+	"url": "onetouchtv.me",
```
Comment (theusaf) [inline websites/O/OneTouchTV/metadata.json:11]: The domain appears to have changed. Perhaps a regex may be useful as well?
```suggestion
	"url": "onetouchtv.cc",
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**